src/cript/api/vocabulary.py

```diff
@@ -0,0 +1,84 @@
+import json
+from typing import Union
+
+import requests
+
+from cript.api.exceptions import InvalidVocabulary
+
+# dictionary of the entire controlled vocabulary
+_ENTIRE_CONTROLLED_VOCABULARY: dict = {}
+
+
+def _get_controlled_vocabulary() -> dict:
+    """
+    gets the entire controlled vocabulary
+    1. checks global variable to see if it is already set
+        if it is already set then it just returns that
+    2. if global variable is empty, then it makes a request to the API
+        and gets the entire controlled vocabulary
+        and then sets the global variable to it
+
+    Returns
+    -------
+    dict
+        controlled vocabulary
+    """
+
+    global _ENTIRE_CONTROLLED_VOCABULARY
+
+    # if cache is empty then make request and set cache
+    if len(_ENTIRE_CONTROLLED_VOCABULARY) == 0:
+        host: str = "https://development.api.criptapp.org"
+        # TODO make request to API to get controlled vocabulary
+        response = requests.get(f"{host}/controlled-vocabulary").json()
+
+        # convert to dict for easier use
+        response = json.loads(response)
+
+        _ENTIRE_CONTROLLED_VOCABULARY = response
+
+    return _ENTIRE_CONTROLLED_VOCABULARY
+
+
+def is_vocab_valid(vocab_category: str, vocab_value: str) -> Union[bool, InvalidVocabulary]:
```

Comment:
This is not supposed to wrap a schema validation function? You have the schema for the vocab, you can check it using your schema library.

Reply:
Yes, that is what is going to happen under the hood there.

Reply:
I was thinking we we're going to get the controlled vocabulary from the API instead of the db schema.

with something like this

```
[
algorithm_key, algorithm_type, building_block, citation_type, computation_type, computational_forcefield_key, ...
]

# for loop to make a request to the api for each end point and put the result inside the _ENTIRE_CONTROLLED_VOCABULARY

```

and when we want to check a vocabulary value like data type, we would call the function with the category and word

```
is_vocab_valid("data_type", "afm_amp")
```